src/pages/index.js

```diff
@@ -119,52 +177,133 @@ function handleEscKey(evt) {
   }
 }
 
+//Edit profile modal listeners
 profileEditButton.addEventListener("click", () => {
   nameInput.value = profileNameElement.textContent;
   jobInput.value = profileJobElement.textContent;
   resetValidation(profileFormElement, [nameInput, jobInput], config);
   openModal(editProfileModal);
 });
 
-newPostButton.addEventListener("click", () => {
-  openModal(newPostModal);
-});
-
 editCloseButton.addEventListener("click", () => {
   closeModal(editProfileModal);
 });
 
+//Edit avatar modal listeners
+editAvatarButton.addEventListener("click", () => {
+  resetValidation(avatarForm, [avatarInput], config);
+  openModal(editAvatarModal);
+});
+
+avatarCloseButton.addEventListener("click", () => {
+  closeModal(editAvatarModal);
+});
+
+//New post modal listeners
+newPostButton.addEventListener("click", () => {
+  openModal(newPostModal);
+});
+
 newPostCloseButton.addEventListener("click", () => {
   closeModal(newPostModal);
 });
 
+//Post preview modal listener
 previewCloseButton.addEventListener("click", () => {
   closeModal(previewModal);
 });
 
+//Card deletion modal listeners
+cardDeleteCloseButton.addEventListener("click", () => {
+  closeModal(cardDeleteModal);
+});
+
+cardDeleteCancelButton.addEventListener("click", () => {
+  closeModal(cardDeleteModal);
+});
+
+//Submission handlers
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Deleting...";
+  api
+    .deleteCard({ _id: selectedCardId })
+    .then(() => {
+      selectedCard.remove();
+      closeModal(cardDeleteModal);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      submitButton.textContent = "Delete";
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Saving...";
+  api
+    .editAvatar(avatarInput.value)
+    .then((data) => {
+      avatarImage.src = data.avatar;
+      closeModal(editAvatarModal);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      submitButton.textContent = "Save";
+    });
+}
+
 function handleProfileFormSubmit(evt) {
   evt.preventDefault();
-  profileNameElement.textContent = nameInput.value;
-  profileJobElement.textContent = jobInput.value;
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Saving...";
+  api
+    .editUserInfo({ name: nameInput.value, about: jobInput.value })
+    .then((data) => {
+      profileNameElement.textContent = data.name;
+      profileJobElement.textContent = data.about;
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      submitButton.textContent = "Save";
+    });
+
   closeModal(editProfileModal);
```

Comment:
`closeModal(editProfileModal);`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**